src/main/java/com/google/jenkins/plugins/computeengine/client/ClientUtil.java

```diff
@@ -73,6 +74,20 @@ private static GoogleRobotCredentials getRobotCredentials(
             ItemGroup itemGroup, List<DomainRequirement> domainRequirements, String credentialsId)
             throws AbortException {
 
+        /* During the integration tests, the parameter `credentialId` is equal to the `<Project-Id>` which is set
+            by the environment variable. But the actual credential created within Jenkins is having `id` as a random
+            UUID. So the `CredentialsMatchers.firstOrNull` was returning `null` due to `CredentialsMatchers.withId
+            (credentialsId)`
```

Comment:
This does not make any sense. https://github.com/jenkinsci/google-compute-engine-plugin/blob/79b8ffdb9a7ad456b7ea013e88802f2ea3cd1046/src/test/java/com/google/jenkins/plugins/computeengine/integration/ITUtil.java#L206 is just using a deprecated constructor overload which fails to pass a specific `credentialsId`. Fix that in tests, and the hack in project main sources should be unnecessary.